src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -513,6 +508,18 @@ private ImmutableArray<NamedTypeSymbol> MakeDeclaredInterfaces()
             }
         }
 
+        private static NamedTypeSymbol DecodeNullableAndTupleTypes(NamedTypeSymbol type, EntityHandle handle, PEModuleSymbol moduleSymbol)
+        {
+            // PROTOTYPE(NullableReferenceTypes): Should call NullableTypeDecoder.TransformOrEraseNullability
+            // here (see StaticNullChecking.UnannotatedAssemblies_09) but TransformOrEraseNullability results in
+            // a StackOverflowException when building Roslyn.sln.
+            if (moduleSymbol.UtilizesNullableReferenceTypes)
+            {
+                type = (NamedTypeSymbol)NullableTypeDecoder.TransformType(TypeSymbolWithAnnotations.Create(type), handle, moduleSymbol).TypeSymbol;
```

Comment:
>(NamedTypeSymbol)NullableTypeDecoder.TransformType(TypeSymbolWithAnnotations.Create(type), handle, moduleSymbol).TypeSymbol; [](start = 23, length = 124)

Dropping top level nullability?

Reply:
This is used for base types and implemented interfaces, which are never null.

---
In reply to: [194567937](https://github.com/dotnet/roslyn/pull/27589#discussion_r194567937) [](ancestors = 194567937)